src/AddImageChecks.cpp

```diff
@@ -13,6 +15,41 @@ using std::pair;
 using std::string;
 using std::vector;
 
+namespace {
+
+// Stores buffer asserts Stmt's ordered by buffer name and assert type.
+class BufferAsserts {
+public:
+    enum Type { DimsNoOverflow,
```

Comment:
Nit: indentation doesn't match that of other enums in Halide, please fix

Reply:
Nit: new enums should probably use `enum class`

Reply:
@steven-johnson BufferAsserts defined in an anonymous namespace we shouldn't have  ambiguity between Halide::Type & BufferAsserts::Type.

Reply:
Ambiguity to the human reader, not to the compiler.